src/03-modality-agnostic-files.md

```diff
@@ -217,3 +217,198 @@ code organization of these scripts at the moment.
+
+iii. If provenance records are included, these records of provenance of a dataset or a file MUST be described using a `[&lt;prefix>_]prov.jsonld` file. Since these [jsonld](https://json-ld.org/) documents are graph objects, they can be aggregated without the need to apply any inheritance principle. 
+
+iv. The provenance file MAY be used to reflect the _provenance of a dataset, a collection of files or a specific file at any level_of the bids hierarchy. 
+
+v. Provenance information SHOULD be anonymized/de-identified as necessary. 
+
+Justification for Separating Provenance from file JSON
+
+Provenance is information about a file, including any metadata that is relevant to the file itself. Thus any BIDS data file and its associated JSON sidecar metadata together constitute a unique entity. As such, one may want to record the provenance of the JSON file as much as the provenance of the BIDS file. In addition, separating the provenance as a separate file for now, allows this to be an OPTIONAL component, and by encoding provenance as a JSON-LD document allows capturing the provenance as an individual record or multiple records distributed throughout the dataset.
+
+Possible places to encode provenance
+
+**Dataset level provenance.** At the dataset level, provenance could be about the dataset itself, or about any entity in the dataset. This provenance may evolve as new data are added, which may include sourcedata, BIDS data, and BIDS derived data. One option is to make use of [named-graphs](https://www.w3.org/TR/json-ld11/#named-graphs).
+
+In this example, with this `prov.jsonld` file we encode that the T1.mgz file was generated by version 6 of the FreeSurfer software.
+
+```json
+{
+  "@context": "https://some/url/to/bids_context.jsonld",
+  "@id": "http://example.org/ds00000X",
+  "generatedAt": "2020-01-10T10:00:00",
+  "wasGeneratedBy": {
+      "@id": "https://banda.mit.edu/",
+      "@type": "Project",
+      "startedAt": "2016-09-01T10:00:00",
+      "wasAssociatedWith": { "@id": "NIH",
+                             "@type": "Organization",
+                             "hadRole": "Funding"
+                           }
+    },
+  "@graph": [
+    {
+      "@id": "sub-01/mri/T1.mgz",
+      "@type": "Image",
+      "sha512": "121231221ab4534...",
+      "derivedFrom": "../sub-01/anat/..._T1.nii.gz",
+      "attributedTo": "MyFreeSurfer",
+      "generatedAt": "2019-01-10T10:00:00"
+    }, 
+    {
+      "@id": "MyFreeSurfer",
+      "@type": "SoftwareAgent",
+      "version": "6.0.0",
+      "RRID": "RRID:SCR_001847"
+    }
+  ]
+}
+```
+
+**File level provenance.** This follows some of the same concepts at the dataset level, but is specifically about the current file under consideration.
+
+```bash
+sub-01/ 
+    func/
+        sub-01_task-xyz_acq-test1_run-1_bold.nii.gz
+        sub-01_task-xyz_acq-test1_run-1_prov.jsonld
```

Comment:
This would appear to apply equally well to `sub-01_task-xyz_acq-test1_run-1_events.tsv`. I would suggest that `prov` is not an appropriate suffix, and could either be made into an extension `.prov` or a double-extension `.prov.jsonld`.

Reply:
@effigies - i agree. i'm waiting for a PR to this PR to get merged before making any additional changes.